### PR TITLE
Add colors option to env.drawboxarray

### PR DESCRIPTION
**Pull request by ompugao**
Add `colors` option to `env.drawboxarray` to specify colors for each box.

### CODE REVIEW

**Comment by ompugao** [comment]
`h = env.drawboxarray(np.random.randn(3000), np.array([0.05,0.05,0.05]), np.random.randn(4000))`

image

**Comment by ompugao** [comment]
ref: #1015

**Comment by cielavenir** [comment]
@rdiankov could you take a look? I didnt know ompugao is JSK people

cc @snozawa @eisoku9618

File: plugins/qtosgrave/qtosgviewer.cpp [hunk 1]
```diff
@@ -1771,19 +1771,24 @@ GraphHandlePtr QtOSGViewer::drawbox(const RaveVector<float>& vpos, const RaveVec
...
         const RaveVector<float>& pos = vpos[i];
         osg::ref_ptr<osg::Box> box = new osg::Box();
         box->setHalfLengths(osg::Vec3(vextents.x, vextents.y, vextents.z));
         box->setCenter(osg::Vec3(pos.x, pos.y, pos.z));
-
+        
         osg::ref_ptr<osg::ShapeDrawable> sd = new osg::ShapeDrawable(box.get());
-        sd->setColor(osg::Vec4f(0.33203125f, 0.5f, 0.898437f, 1.0f));
+        if (size_mismatch) {
+            const RaveVector<float>& color = vcolors[0];
```
Comment (rdiankov) [inline plugins/qtosgrave/qtosgviewer.cpp]: @ompugao Is `vcolors` guaranteed to have elements in it?

Comment (ompugao) [reply plugins/qtosgrave/qtosgviewer.cpp]: Thank you for your review. No, it can be invalid memory access when called from c++.
Fixed.

**Comment by rdiankov** [comment]
woot, thanks for the change!

**Comment by ompugao** [comment]
thank you for merging.

**Pull request merged**